Azure Services/Azure Monitor/Agents/Migration Tools/DCR Config Generator/VMIDcrGeneratorModule/VMIDcrGeneratorModule.psm1

```diff
@@ -0,0 +1,194 @@
+
+function Find-IfVmiEnabled
+{
+    param (
+        [Parameter(Mandatory=$true)][string] $ResourceGroupName,
+        [Parameter(Mandatory=$true)][string] $WorkspaceName
+    )
+    
+    $result = Get-AzOperationalInsightsIntelligencePack -ResourceGroupName $ResourceGroupName -WorkspaceName $WorkspaceName | Where-Object {$_.name -match ".*VMInsights*" }
+    if ($result -and $result.Enabled -eq "True") {

```

Comment:
If an empty workspace is provided to the script, it was earlier expected that the output would be empty, but rather from the experiment.

PS /home/venkata> $result = Get-AzOperationalInsightsIntelligencePack -ResourceGroupName emptyworkspace -WorkspaceName emptyworkspace1 | Where-Object {$_.name -match ".*VMInsights*" }
PS /home/venkata> Write-Output $result             
Name       DisplayName                 Enabled
----       -----------                 -------
VMInsights Azure Monitor - VM Insights   False

PS /home/venkata>